### DISCUSSION TITLE
Lesson 11: test on rinkeby no longer working

### DISCUSSION THREAD

**Discussion by Karlus44** [smartcontractkit/full-blockchain-solidity-course-py#1425] [Q&A] [answered]
Hello,
I've just completed the lesson 11, but I tried my tests again, and the test_can_create_advanced_collectible_integration on rinkeby doesn't work, though I'm pretty sure it worked before.
The assert advanced_collectible.tokenCounter() == 1 because the tokenCounter value has not been updated (it's equal to 0).
If you need to check my code it's here: https://github.com/Karlus44/nft-demo , the test failing is in the tests/integration directory.
My contract seems to work correctly and can be checked here: 0x2a3ef6be87c5e39208fb151cb5dfbdede95e22d8 on rinkeby.etherscan
Thanks for any help !

**Comment by cromewar** [comment, marked as answer]
Hello @Karlus44 
I'll give a look to the code, but on the meantime could you try running the code again with `s` flag and share the complete error log? Something like this:
`brownie test -s -k test_name`
Cheers.

**Reply by Karlus44** [reply]
Hello, I launched it a few hours ago now and it's still running. I've never tested the -s option. Does it usually take so long ?

**Reply by Karlus44** [reply]
It was very long but indeed this was actually working. Here is the complete error log:

```
karlus@galois:~/Documents/NFT-Contract/nft-demo$ brownie test -s -k test_can_create_advanced_collectible_integration --network rinkeby
Brownie v1.18.1 - Python development framework for Ethereum

========================== test session starts ===========================
platform linux -- Python 3.8.10, pytest-6.2.5, py-1.11.0, pluggy-1.0.0 --
/home/karlus/.local/pipx/venvs/eth-brownie/bin/python
cachedir: .pytest_cache
hypothesis profile 'brownie-verbose' -> verbosity=2, deadline=None, max_examples=50, stateful_step_count=10, report_multiple_bugs=False, database=DirectoryBasedExampleDatabase(PosixPath('/home/karlus/.brownie/hypothesis'))

rootdir: /home/karlus/Documents/NFT-Contract/nft-demo
plugins: eth-brownie-1.18.1, forked-1.4.0, web3-5.27.0, hypothesis-6.27.3, xdist-1.34.0
collected 3 items / 2 deselected / 1 selected

tests/integration/test_advanced_collectible_integration.py::test_can_create_advanced_collectible_integration
RUNNING
Funded 0x305D14c36039dA8cE04FC6a8c5346D9416502893!
New token has been created!
tests/integration/test_advanced_collectible_integration.py::test_can_create_advanced_collectible_integration FAILED

=============================================== FAILURES ================================================
___________________________ test_can_create_advanced_collectible_integration ____________________________

    def test_can_create_advanced_collectible_integration():
        # deploy the contract
        # create an NFT
        # get a random breed back
        # Arrange
        if network.show_active() in LOCAL_BLOCKCHAIN_ENVIRONMENTS:
            pytest.skip()
        # Act
        advanced_collectible, creation_transaction = deploy_and_create()
        time.sleep(60)

        # Assert
>       assert advanced_collectible.tokenCounter() == 1
E       assert 0 == 1
E         +0
E         -1

tests/integration/test_advanced_collectible_integration.py:19: AssertionError
tests/integration/test_advanced_collectible_integration.py::test_can_create_advanced_collectible_integration ======================================== short test summary info ========================================
FAILED tests/integration/test_advanced_collectible_integration.py::test_can_create_advanced_collectible_integration
============================= 1 failed, 2 deselected in 14458.97s (4:00:58) =============================
```

**Reply by cromewar** [reply]
Did you remember to add plus one to the token counter on each call?

**Reply by Karlus44** [reply]
yes, as you can see in my contract, https://github.com/Karlus44/nft-demo/blob/main/contracts/AdvancedCollectible.sol ,
the tokenCounter is incremented each time my function fulfillRandomness is called.

**Reply by cromewar** [reply]
I'll give it a check, but also remember that fulfill randomness is an asynchronous call, so the Breed count could take some time to get complete, try with a tx.wait(180) and see what happens

**Reply by Karlus44** [reply]
yes, I had a time.sleep(60). But now with a time.sleep(180) instead I passed the test.
Thank you, you solved my case !

**Reply by cromewar** [reply]
I'm glad is solved now!.
